src/main/java/com/google/codeu/servlets/StatsPageServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.codeu.data.Datastore;
+import com.google.gson.JsonObject;
+/**
+ * Handles fetching site statistics. 
+ * 
+ * This uses the init() function that creates and store a Datastore instance,
+ * then doGet() calls the function getTotalMessageCount() in the Datastore.java
+ * and create a JSON that countains the message count
+ */
+public class StatsPageServlet extends HttpServlet{
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
```

Comment:
Don't forget to call `super.init();` -- you don't want to accidentally omit necessary initialization of the base class.